insights/util/spec_processor.py

```diff
@@ -0,0 +1,399 @@
+import logging
+import os
+import re
+import six
+import socket
+import struct
+
+from tempfile import TemporaryFile
+
+logger = logging.getLogger(__name__)
+
+# The regex for password removal, which is read from the "/etc/insights-client/.exp.sed"
+default_password_regexs = [
+    "(password[a-zA-Z0-9_]*)(\s*\:\s*\"*\s*|\s*\"*\s*=\s*\"\s*|\s*=+\s*|\s*--md5+\s*|\s*)([a-zA-Z0-9_!@#$%^&*()+=/-]*)",
+    "(password[a-zA-Z0-9_]*)(\s*\*+\s+)(.+)",
+]
+
+
+class PostProcessor(object):
```

Comment:
Since it also does some `Redaction` stuff, I have changed my mind about naming it as `PostObfuscation` initially. And about the `Post`, I think it's a bit suitable to indicate these `processes` are done at the end of the collection. Or how do you think about `PostCleaner`,  `InsightsSpecsCleaner`,  or `InsightsArchiveCleaner` (from the soscleaner)

Reply:
Furthermore, I think we should also move the process of `"filters"` into this module, how do you think?

Reply:
@ryan-blakley - regarding the naming. we think "spec_cleaner" would be more suitable per our team discussion. So I went ahead and renamed it as "spec_cleaner.Cleaner" and moved it from the `insights/util/spec_process.py` to the `insights/core/spec_cleaner.py`. Please kindly review again.

Reply:
I got to thinking over the weekend that since the redaction was moved in there the obfuscation name might not be the best, my initial thought about InsightsCleaner.  I think naming it Cleaner is even cleaner ;) and works for me.